File: lab2/src/objects/MyVector.scala
package objects

import java.util.Comparator
import List.IUserType



class MyVector(xCordA: Integer =(Math.random * 100).toInt - 100,yCordA: Integer = (Math.random * 100).toInt - 100,
               xCordB: Integer = (Math.random * 100).toInt - 100,
               yCordB: Integer = (Math.random * 100).toInt - 100 ) extends IUserType with Serializable {

  var y = this.yCordB - this.yCordA
  var x = this.xCordB - this.xCordA


  def getA: String = {
    val str = "(" + xCordA + ";" + yCordA + ")"
    str
  }

  def getB: String = {
    val str = "(" + xCordB + ";" + yCordB + ")"
    str
  }

  override def toString: String = "(" + this.x + ";" + this.y + ")"

  override def copy = new MyVector(this.xCordA, this.yCordA, this.xCordB, this.yCordB)

  override def create = new MyVector

  override def getClassName: String = this.getClass.getName

  override def getTypeComparator = new SortMyVector

  override def getCompare: Double = {
    val sqrt = Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2))
    val length = Math.abs(sqrt)
    length
  }
}

File: lab1/src/objects/MyVector.java
package objects;

import java.util.Comparator;

import kitpo.IUserType;
import java.io.Serializable;
import java.util.Comparator;

import kitpo.IUserType;
public class MyVector implements IUserType,Serializable{

    private final Integer xCordA;
    private final Integer yCordA;
    private final Integer xCordB;
    private final Integer yCordB;
    private final Integer x;
    private final Integer y;


    public String getA() {
        String  str = "(" + xCordA + ";" + yCordA + ")";
        return str;
    }

    public String getB() {
        String  str = "(" + xCordB + ";" + yCordB + ")";
        return str;
    }

    public double getLength() {
        double sqrt = Math.sqrt(Math.pow(this.x,2)+Math.pow(this.y,2));
        double length = Math.abs(sqrt);
        return length;
    }

    public MyVector(Integer xA,Integer yA,Integer xB,Integer yB) {
        this.xCordA = xA;
        this.yCordA = yA;
        this.xCordB = xB;
        this.yCordB = yB;
        this.y = this.yCordB - this.yCordA;
        this.x = this.xCordB - this.xCordA;

    }

    public MyVector() {
        this.xCordA = (int) (Math.random() * 100) - 100;
        this.yCordA = (int) (Math.random() * 100) - 100;
        this.xCordB = (int) (Math.random() * 100) - 100;
        this.yCordB = (int) (Math.random() * 100) - 100;
        this.y = this.yCordB - this.yCordA;
        this.x = this.xCordB - this.xCordA;
    }

    @Override
    public String toString() {
        return "(" + this.x + ";" + this.y + ")";
    }

    @Override
    public IUserType copy() {
        return new MyVector(this.xCordA, this.yCordA,this.xCordB,this.yCordB);
    }

    @Override
    public IUserType create() {
        return new MyVector();
    }

    @Override
    public String getClassName() {
        return this.getClass().getName();
    }

    @Override
    public Comparator getTypeComparator() {
        return new SortMyVector();
    }
}
